app/api/v1/endpoints/chat_session_endpoints.py

```
@@ -0,0 +1,505 @@
+@router.post("/chat-sessions", response_model=ChatSession, tags=["Chat Sessions"])
+async def create_chat_session(
+    session_data: ChatSessionCreate,
+    current_user: TokenPayload = Depends(get_current_user)
+):
+    """Create a new chat session"""
+    try:
+        session_dict = create_chat_session_in_db(session_data, current_user.sub)
+        return ChatSession(**session_dict)
+    except psycopg2.Error as e:
+        logger.error(f"Database error creating chat session: {e}")
+        raise HTTPException(status_code=500, detail="Error creating chat session")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)